tests/post/steps/test_service_configuration.py

```diff
@@ -158,6 +160,21 @@ def _wait_for_deployment():
                 value, service, utils.get_dict_element(response, path)
             )
         )
+        # original ConfigMap returned from Given block will be used to
+        # patch back the ConfigMap to it's original state
+        original_name = check_csc_configuration['name']
+        original_namespace = check_csc_configuration['namespace']
+        original_csc_obj = check_csc_configuration['csc_obj']
+
+        patch = {
+            'data': {
+                'config.yaml': yaml.safe_dump(
+                    original_csc_obj, default_flow_style=False
+                )
+            }
+        }
+        csc.update(original_name, original_namespace, patch)
```

Comment:
I would prefer to do that in a teardown function
Because here if for what ever reason we want to add another `then` check .... that will not work as after this `then` you reset the csc

Reply:
Maybe we need an explicit `then` step, like `Then we restore the original state of the config map foo in namespace bar`.
It may be annoying, but at least it's explicit;

Otherwise, yeah we have to assume that this `then` comes last or that we won't have more `then` checks on the modified object after this one.

Reply:
Hmm adding this "restore `then`" look good and clean to me

Reply:
ok